human.js

```diff
@@ -0,0 +1,73 @@
+class Human {
+  constructor(name, age, dateOfBirth) {
+    this.name = name;
+    this.age = age;
+    this.dateOfBirth = dateOfBirth;
+  }
+
+  displayInfo() {
+    let result = "";
+    for (const key in this) {
+      if (this.hasOwnProperty(key)) {
+        result += key + ": " + this[key] + " ";
+      }
+    }
+    return result;
+  }
+}
+
+class Employee extends Human {
+  constructor(name, age, dateOfBirth, salary, department) {
+    super(name, age, dateOfBirth);
+    this.salary = salary;
+    this.department = department;
+  }
+
+  displayInfo() {
+    return super.displayInfo(); // + `salary: ${this.salary}, department: ${this.department}`;
+  }
+}
+
+class Manager extends Employee {
+  constructor(...args) {
+    super(...args);
+    this.developers = [];
+  }
+
+  addDeveloper(developer) {
+    this.developers.push(developer);
+    developer.setMamager = this;
+  }
+
+  removeDeveloper(developer) {
+    this.developers.splice(this.developers.indexOf(developer), 1);
```

Comment:
```suggestion
    this.developers.splice(this.developers.indexOf(developer), 1);
```
а вдруг разработчика не будет в массиве, тогда `indexOf()` вернет `-1`, в итоге из массива удалится последний элемент

Reply:
Действительно..